src/Compilers/Core/Portable/Syntax/SyntaxToken.cs

```diff
@@ -634,9 +634,12 @@ internal SyntaxToken GetPreviousToken(Func<SyntaxToken, bool> predicate, Func<Sy
         /// </summary>
         public Location GetLocation()
         {
-            return Node != null
-                ? this.SyntaxTree.GetLocation(this.Span)
-                : Location.None;
+            if (this.Node == null || this.SyntaxTree == null)
```

Comment:
>this.Node == null [](start = 16, length = 17)

Would the check for SyntaxTree be sufficient on its own, or do we really need to check this condition as well?